src/Build.UnitTests/BackEnd/BuildManager_Tests.cs

```diff
@@ -4093,6 +4093,32 @@ public void IdenticalSubmissionsShouldCompleteAndNotHangTheBuildOnMissingTargetE
             }
         }
 
+        [Fact]
+        public void BuildWithZeroConnectionTimeout()
+        {
+            string contents = CleanupFileContents(@"
+<Project xmlns='msbuildnamespace' ToolsVersion='msbuilddefaulttoolsversion'>
+ <Target Name='test'>
+    <Message Text='Text'/>
+ </Target>
+</Project>
+");
+            _env.SetEnvironmentVariable("MSBUILDNODECONNECTIONTIMEOUT", "0");
+
+            BuildRequestData data = GetBuildRequestData(contents);
+            try
+            {
+                BuildResult result = _buildManager.Build(_parameters, data);
+
+                // The build should either finish successfully (very unlikely).
+                result.OverallResult.ShouldBe(BuildResultCode.Success);
+            }
+            catch (InternalErrorException)
+            {
+                // Or it should throw InternalErrorException because the node didn't get connected within 0ms.
+            }
```

Comment:
```suggestion
            catch (Exception e)
            {
                // Or it should throw InternalErrorException because the node didn't get connected within 0ms.
                e.ShouldBeOfType<InternalErrorException>();
            }
```

?